### PR TITLE
Add Directory to Dashboard

### PR DESCRIPTION
**Pull request by OR13**
The Directory provides public / key server style information about a user.

It supports a simple list and detail view.

### CODE REVIEW

**Comment by OR13** [comment]
I can think of some improvements to this immediately, but this PR represents an initial implementation.

In particular the styling / design of these directory pages could be drastically improved.

File: packages/transmute-dashboard/src/components/Directory/DirectoryPage.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      <AppBar>
+        <DirectoryTable people={this.state.profiles} />
+      </AppBar>
+    );
+  }
+}
+
+const mapDispatchToProps = dispatch => {
+  return {
+    // setUserProfile: async (auth, profile) => {
```
Comment (eolszewski) [inline packages/transmute-dashboard/src/components/Directory/DirectoryPage.js]: @OR13 remove unused.

File: packages/transmute-dashboard/src/components/Directory/DirectoryPage.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    if (!this.state.profiles) {
+      const profiles = await getDirectoryProfiles(this.props.auth);
+      this.setState({
+        profiles
+      });
+    }
+  }
+
+  render() {
+    // const { classes } = this.props;
```
Comment (eolszewski) [inline packages/transmute-dashboard/src/components/Directory/DirectoryPage.js]: remove

File: packages/transmute-dashboard/src/components/Directory/Profile/DirectoryProfilePage.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    return (
+      <AppBar>
+        <h1>Profile</h1>
+        <pre>{JSON.stringify(profile, null, 2)}</pre>
+      </AppBar>
+    );
+  }
+}
+
+const mapDispatchToProps = dispatch => {
```
Comment (eolszewski) [inline packages/transmute-dashboard/src/components/Directory/Profile/DirectoryProfilePage.js]: Just delete this and replace with 'null' in the connect function if it is unused.

File: packages/transmute-dashboard/src/components/Routes/index.js [hunk 4]
```diff
@@ -34,29 +36,16 @@ class Routes extends React.Component {
...
-            path="/implicit/callback"
-            component={ImplicitCallback}
-          />
+          <Route path="/" exact component={Home} />
+          <Route path="/demo" component={Demo} />
+          <Route path="/register" exact render={() => <Register />} />
+          <Route path="/implicit/callback" component={ImplicitCallback} />
+          <SecureRoute path="/profile" exact render={() => <Profile />} />
+          <SecureRoute path="/directory" exact render={() => <Directory />} />
           <SecureRoute
```
Comment (eolszewski) [inline packages/transmute-dashboard/src/components/Routes/index.js]: If you're going to update formatting of the routes, update all of the routes.

**Comment by OR13** [comment]
I have seen these change requests, and will be implementing them when I get a chance.

**Comment by OR13** [comment]
I'm going to reduxify this

File: packages/transmute-dashboard/src/components/Directory/DirectoryTable/DirectoryTable.js [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+const toolbarStyles = theme => ({
+  root: {
+    paddingRight: theme.spacing.unit
+  },
+  title: {
+    flex: '0 0 auto'
+  }
+});
+
+let EventStoresTableToolbar = props => {
```
[the remark this answers is not in the document]
Comment (eolszewski) [reply packages/transmute-dashboard/src/components/Directory/DirectoryTable/DirectoryTable.js]: Still need to fix the naming of anything with 'eventstore' in it to reflect the new class.

File: packages/transmute-dashboard/src/components/Routes/index.js (hunk 4 above)
Comment (eolszewski) [reply packages/transmute-dashboard/src/components/Routes/index.js]: Same comment

**Comment by OR13** [comment]
good catch on those names, totally missed that.

**Pull request merged**